src/dos/cdrom_image.cpp

```diff
@@ -223,39 +369,130 @@ bool CDROM_Interface_Image::GetMediaTrayStatus(bool& mediaPresent, bool& mediaCh
 	mediaPresent = true;
 	mediaChanged = false;
 	trayOpen = false;
+
+#ifdef DEBUG
+	LOG_MSG("%s CDROM: GetMediaTrayStatus present=%d, changed=%d, open=%d", get_time(), mediaPresent, mediaChanged, trayOpen);
+#endif
+
 	return true;
 }
 
-bool CDROM_Interface_Image::PlayAudioSector(unsigned long start,unsigned long len)
+bool CDROM_Interface_Image::PlayAudioSector(unsigned long start, unsigned long len)
 {
-	// We might want to do some more checks. E.g valid start and length
-	SDL_mutexP(player.mutex);
-	player.cd = this;
-	player.currFrame = start;
-	player.targetFrame = start + len;
-	int track = GetTrack(start) - 1;
-	if(track >= 0 && tracks[track].attr == 0x40) {
+	bool is_playable(false);
+	const int track = GetTrack(start) - 1;
+
+	// The CDROM Red Book standard allows up to 99 tracks, which includes the data track
+	if ( track < 0 || track > 99 )
+		LOG(LOG_MISC, LOG_WARN)("Game tried to load track #%d, which is invalid", track);
+
+	// Attempting to play zero sectors is a no-op
+	else if (len == 0)
+		LOG(LOG_MISC, LOG_WARN)("Game tried to play zero sectors, skipping");
+
+	// The maximum storage achieved on a CDROM was ~900MB or just under 100 minutes
+	// with overburning, so use this threshold to sanity-check the start sector.
+	else if (start > 450000)
+		LOG(LOG_MISC, LOG_WARN)("Game tried to read sector %lu, which is beyond the 100-minute maximum of a CDROM", start);
+
+	// We can't play audio from a data track (as it would result in garbage/static)
+	else if(track >= 0 && tracks[track].attr == 0x40)
```

Comment:
Is there some enum indicating 0x40 to be a data track?

Reply:
0x40 identifies a data track (0x41 is a lead-in track, and others). There unfortunately isn't an enum set locking down these magic identifiers, but there should be (DOSBox is riddled with unexplained, unreferenced magic numbers just like this).

Unfortunately I'm similarly worried about "change overreach" if I start changing more files (cdrom.cpp, cdrom_aspi_win32.cpp, and cdrom_image.cpp) than are needed to impart the feature's I'm added.

Sigh.  So much fear and uncertainty because of essentially little or no communication.  Hopefully this gets cleared up soon so the pathway is clear - one way or another.

Reply:
Understood. I agree - let's get this merged before branching out to improvements in other areas.